### DISCUSSION TITLE
broadcastQueryClient does not propagate invalidations

### DISCUSSION THREAD

**Discussion by robert-theodo** [TanStack/query#2591] [General]
We are trying to use broadcastQueryClient on our website.

**Use Case**
We have two browser tabs. One tab is a list of items. The second tab is a more detailed editor for one item from the list. When we save the item from the editor, we want to invalidate the query for the list on the list tab so that edited attributes get updated. 

**Problem**
It looks like only updates are propagated through the broadcast client, not invalidations.

How can we achieve our use case? Any pointers on how to open a PR for the broadcastQueryClient?

**Comment by TkDodo** [comment]
I think the broadcastQueryClient plugin just assumes the same queryClients in all tabs. So you'd need to invalidate in your tab, fetch the updates, and then it should sync to the other tabs because of the `queryUpdated` event:

https://github.com/tannerlinsley/react-query/blob/4bbfe2ab97b83a40553e7c78c6174907cb708682/src/broadcastQueryClient-experimental/index.ts#L35-L45

If you'd send an `invalidate`  event or so to the other tab, and it would re-fetch, then syncing would just post the updates back to the first tab anyhow, so I think you can just do the actual invalidation in the same tab as well. But I've never really worked with the plugin, so I could be wrong...
